### PR TITLE
feat(kuma-cp): introduce OpenTelemetry tracing

### PR DESCRIPTION
**Pull request by michaelbeaumont**
```
Span #1
    Trace ID       : ce74b2c0cb19c2bf5dc1eddb13b075c2
    Parent ID      :
    ID             : cc2c6e954782a543
    Name           : /meshes
    Kind           : SPAN_KIND_SERVER
    Start time     : 2023-06-30 13:05:27.023127684 +0000 UTC
    End time       : 2023-06-30 13:05:27.023294989 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message :
Attributes:
     -> http.method: STRING(GET)
     -> http.scheme: STRING(http)
     -> http.flavor: STRING(1.1)
     -> net.host.name: STRING(api-server)
     -> net.host.port: INT(5681)
     -> net.sock.peer.addr: STRING(::1)
     -> net.sock.peer.port: INT(44582)
     -> http.user_agent: STRING(HTTPie/3.2.1)
     -> http.route: STRING(/meshes)
     -> http.status_code: INT(200)
```

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues -- 
  https://github.com/kumahq/kuma/issues/5696
  https://github.com/kumahq/kuma-website/issues/1397
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  none
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/config/tracing/config.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package tracing
+
+type Config struct {
+	OpenTelemetry OpenTelemetry `json:"openTelemetry"`
+}
+
+type OpenTelemetry struct {
+	// Address of OpenTelemetry collector.
+	// E.g. otel-collector:4317
+	Endpoint string `json:"endpoint"`
```
Comment (jakubdyszkiewicz) [inline pkg/config/tracing/config.go:10]: envconfig?

Comment (jakubdyszkiewicz) [reply pkg/config/tracing/config.go:10]: and add to kuma cpc defaults

File: pkg/config/app/kuma-cp/config.go [hunk 2]
```diff
@@ -150,6 +151,8 @@ type Config struct {
 	Proxy xds.Proxy `json:"proxy"`
 	// Intercommunication CP configuration
 	InterCp intercp.InterCpConfig `json:"interCp"`
+	// Tracing
+	Tracing *tracing.Config `json:"tracing"`
```
Comment (jakubdyszkiewicz) [inline pkg/config/app/kuma-cp/config.go:155]: nit: why pointer

Comment (michaelbeaumont) [reply pkg/config/app/kuma-cp/config.go:155]: yeah I can check if otel.endpoint is empty instead

**Review by lahabana — APPROVED** [review]

**Review by jakubdyszkiewicz — APPROVED** [review]

**Review by frzifus — APPROVED** [review]
lgtm :)

File: pkg/plugins/runtime/opentelemetry/plugin.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+
+	conn, err := grpc.DialContext(
+		ctx,
+		otelConfig.Endpoint,
+		grpc.WithTransportCredentials(
+			insecure.NewCredentials(),
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/plugins/runtime/opentelemetry/plugin.go:89]: yeah we need to add this in another PR

[the remark this answers is not in the document]
Comment (slonka) [reply pkg/plugins/runtime/opentelemetry/plugin.go:89]: @michaelbeaumont - do we have an issue for that?

**Pull request merged**